### DISCUSSION TITLE
How to turn off "Profile Imported from AAPS" notification sound?

### DISCUSSION THREAD

**Discussion by parapenT1sta** [NightscoutFoundation/xDrip#2092] [General]
Since I updated to the latest version with AAPS profile import, I'm getting an popup with the "Profile Imported from AAPS" and I'm trying to find the option to disable the notification sound but I can't find it in any alarms menu or InterApps menu.

How to silence this popup?

**Comment by Navid200** [comment]
@jamorham

**Reply by jamorham** [reply]
Good to know that the feature is working. I will add a preference switch to disable the notification sound in the next nightly. Can you tell me when this sound happens? Is it only when activating a profile switch manually in AAPS or does it occur at other times? Thanks.

**Reply by parapenT1sta** [reply]
Thank you for your reply, in fact happens several times during the day but I cannot tell for what actions are working. That's why I am getting many sound notifications and I would like to mute them. Disabling notifications in Android doesn't help.

**Reply by kenkolden** [reply]
Nightly build 28th Apr 2022
Alarm occurs after updating basal profile in AAPS and at random times during the day.
A popup occurs at bottom of screen when this happens but no notification in notification bar unable to disable alarm
The sound is very annoying and ignores silent mode setting in phone

**Reply by izsapir** [reply]
Hi, im getting the same ,can you pls explain as well what is this featue?

**Reply by LewEasy** [reply]
I get the notification aswell. I'm not sure if it is only with profile switches (manually or enacted by automations) or if there are any other triggers. I'll keep an eye out.

It stops when I disable "accept treatments". But in the long run, I'd like to reactivate that feature without being acoustically notified of profile switches.

**Reply by phrogg** [reply]
For me, it happens about every five minutes actually and is quite annoying :P

**Reply by Navid200** [reply]
@phrogg This is the solution:  https://navid200.github.io/xDrip/docs/AAPS_ProfileImportNotification.html
After you try, if it doesn't work, please let us know.

**Reply by phrogg** [reply]
> @phrogg This is the solution: https://navid200.github.io/xDrip/docs/AAPS_ProfileImportNotification.html After you try, if it doesn't work, please let us know.

Already did thanks, I will reply back if the problem persists.

**Comment by psonnera** [comment]
Fixed here https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.04.30 but there is probably something in NSClient because the toast message appears frequently even if no profile switch is done. Not knowing where to look in debug (probably need to enable something to see that) I think that either xDrip+ triggers profile update upon any NSClient broadcast or AAPS broadcasts the full profile every treatment broadcast.

**Reply by dar-kod** [reply]
Yes, I confirm that sound notification appears when no profile switch is done.

**Reply by dar-kod** [reply]
But this version https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.04.30 doesn't fix this in my case. I can still hear the sound notification, and sometimes very often.

**Reply by psonnera** [reply]
Just disable them https://xdrip.readthedocs.io/en/latest/use/interapp/#receiving-from-nsclient

**Reply by dar-kod** [reply]
Thank you.

**Reply by hartlijn** [reply]
Disabling unfortunately still triggers a high frequent vibration, also with Import sound checkbox off. I can't find a way to disable this without turning vibration off for my Alerts. Anyone?

**Reply by psonnera** [reply]
Something with the notification channel maybe? https://xdrip.readthedocs.io/en/latest/use/alarms/#notification-channel

**Reply by hartlijn** [reply]
Thought about that as well. I have the setting OFF for notification channel, but Android does show me the channels. Confusing.. ?

**Reply by psonnera** [reply]
@Navid200 ?

**Reply by Navid200** [reply]
Something has changed recently.  I get a vibration for my forecast low and persistent high alerts, which I never did in the past.
We need to figure out exactly which release has caused it.

@hartlijn Can you test the May 6, 2022 release to see if it behaves the same way or not?
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.05.06

**Reply by Navid200** [reply]
>  I can't find a way to disable this without turning vibration off for my Alerts. 

How do you do that?

**Reply by hartlijn** [reply]
> Something has changed recently. I get a vibration for my forecast low and persistent high alerts, which I never did in the past. We need to figure out exactly which release has caused it.
> 
> @hartlijn Can you test the May 6, 2022 release to see if it behaves the same way or not? https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.05.06

Yes. In both 2022.05.06 and 2022.05.26 nightly, vibration remains on but sound can be switched off with the new setting.

I'd like to not receive the persistent profile import update at all. No sound, no vibration and no toast pop-up.

**Comment by icnich** [comment]
I've got the sounds turned off, which is great. I'm also experiencing the incessant "xDrip imported AAPS profile" pop up though. Has anyone found a way to disable this?

**Reply by Navid200** [reply]
I have opened  a PR to change the toast message to a log message:
https://github.com/NightscoutFoundation/xDrip/pull/2175

**Reply by icnich** [reply]
Thank you! I had seen another message of you requesting a screenshot, but don't see it here. Would that still be helpful?

I'm quite new to all this, but can I test the change you outlined in the PR? Would that also be helpful to help determine if your tweaks worked? How would I go about that?

Thank you very very much!

**Reply by Navid200** [reply]
No, you cannot test unfortunately.
The only way to test is to compile it and then, you will have to uninstall first, which is not a good idea at all.
But, this is a very simple change.

The head developer needs to know that the users are annoyed by the frequent pop ups.
  
He also may have a reason wanting to keep it as a toast rather than a log.  We will only know that when he gets to see my proposal.

**Reply by icnich** [reply]
Alright! Thank you!

**Comment by Navid200** [comment]
The notification is gone now (June 12, 2022). 

If you  want to be notified for every profile update, you will need to enable the sound.
But, the visual notification is gone now.  
The only enforced record is now a log.

**Reply by icnich** [reply]
Wonderful news, thank you so much!

**Reply by Navid200** [reply]
https://navid200.github.io/xDrip/docs/AAPS_ProfileImportNotification.html
